modules/terraform/validate_test.go

```diff
@@ -0,0 +1,39 @@
+package terraform
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitAndValidateWithNoError(t *testing.T) {
+	t.Parallel()
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	options := &Options{
+		TerraformDir: testFolder,
+	}
+
+	InitAndValidate(t, options)
+	out, err := ValidateE(t, options)
+	require.NoError(t, err)
+	require.Contains(t, out, "Success! The configuration is valid")
+}
+
+func TestValidateWithoutInitWithError(t *testing.T) {
+	t.Parallel()
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	options := &Options{
+		TerraformDir: testFolder,
+	}
+
+	out, err := ValidateE(t, options)
+	require.Error(t, err)
+	require.Contains(t, out, "Plugin reinitialization required. Please run \"terraform init\"")
```

Comment:
This doesn't seem like the case we're most interested in... Instead, we should add a test fixture with _invalid_ Terraform code, call `InitAndValidate` on it, and ensure we get back an error and the expected text.

Reply:
Agreed. 
How about using the `terraform-with-plan-error` test fixture for this case, instead of creating a new one? 
It's basically  a test case that should indicate the functionality we are looking for, since it is trying to output an undeclared input variable.

Reply:
Perfect, thanks!